feat: add MySQL command support PORT

bgajjala8 opened the pull request:
## Description
Porting over a PR from an open source contribution to trigger testing suite.

## PCI review checklist
<!-- heimdall_github_prtemplate:grc-pci_dss-2024-01-05 -->
- [X] I have documented a clear reason for, and description of, the change I am making.
- [X] If applicable, I've documented a plan to revert these changes if they require more than reverting the pull request.
- [X] If applicable, I've documented the impact of any changes to security controls.
  Examples of changes to security controls include using new access control methods, adding or removing logging pipelines, etc.

bgajjala8 commented:
@johanbrandhorst
It looks like two tests were added. I've addressed the linting issues for the tests in the `base_plus `directory. Can we remove the duplicate test from the `base` directory? (I'm assuming it's a copy-paste error.)

johanbrandhorst commented on testing/internal/e2e/tests/base_plus/target_tcp_connect_mysql_test.go:
```diff
@@ -0,0 +1,125 @@
+
+	var buf bytes.Buffer
+	_, _ = io.Copy(&buf, f)
+	require.Contains(t, buf.String(), "mysql>", "Session did not return expected MySQL prompt")
```
Looks like this is [failing today](https://github.com/hashicorp/boundary/actions/runs/16011194421/job/45216827194?pr=5891) in CI:
```
ERROR 2013 (HY000): Lost connection to server at 'handshake: reading initial communication packet', system error: 11\r\n\a
```
I was able to get this working locally on my machine, so we will need to try to debug why it's not working in CI. I can't tell what's going on based on the logs, so we should probably add some more debug logging to this test and the rerun CI to see if we can see what's going on. Since we're having connectivity issues, it would be nice if we could add some more logging around the attempt to establish a connection on line 50. I would try and run some of these commands through that and print them to see if that works. If so, it might be a problem with the Boundary mysql command. To speed up iteration, I would recommend trying to run it locally first and then push it up to CI. Let me know if you need help figuring out how to run the e2e tests locally.

johanbrandhorst replied:
Also, I think we deleted the wrong file, this test should be in the `base` folder, not the `base_plus` folder.

bgajjala8 replied:
Thanks for catching that, I got it mixed up 🙃 - updated

bgajjala8 commented on internal/cmd/commands/connect/mysql.go:
```diff
@@ -0,0 +1,120 @@
+		}
+	}
+	return
+}
```
```suggestion
}

```
New line

bgajjala8 replied:
```suggestion
}

```

bgajjala8 commented on testing/internal/e2e/tests/base/target_tcp_connect_mysql_test.go:
```diff
@@ -0,0 +1,146 @@
+	mysqlPort = "3306"
+
+	return mysqlUser, mysqlPassword, mysqlDb, networkAlias, mysqlPort
+}
```
```suggestion
}

```
new line

bgajjala8 commented:
@johanbrandhorst  
I went ahead and updated tests to work with our E2E suite. I also cleaned up how database variables are declared, so they're less explicit now. Would love a re-review on the PR!

johanbrandhorst commented on testing/internal/e2e/infra/docker.go:
```diff
@@ -303,3 +303,44 @@ func StartOpenSshServer(t testing.TB, pool *dockertest.Pool, network *dockertest
+
+	return &Container{
+		Resource:     resource,
+		UriLocalhost: "http://localhost:3306",
+		UriNetwork:   "http://e2emysql:3306",
```
It's neat that we don't have to provide credentials to this function anymore, but this makes the container harder to use (the user has to guess to look at the environment variables). It looks like MySQL [supports using URI connection strings](https://dev.mysql.com/doc/refman/8.4/en/connecting-using-uri-or-key-value-pairs.html#connecting-using-uri) in the same way as postgres does. What do you think about putting the credentials in these returned strings instead and extract them out in the caller?
```suggestion
		UriLocalhost: fmt.Sprintf("mysql://%s:%s@localhost:3306/%s", mysqlUser, mysqlPassword, mysqlDb),
		UriNetwork:   fmt.Sprintf("mysql://%s:%s@%s:3306/%s", mysqlUser, mysqlPassword, networkAlias, mysqlDb),
```

In the test we can extract this using `url.Parse`:

```go
u, err := url.Parse(container.UriLocalhost)
if err != nil {
...
}
user := u.User.Username()
pw := u.User.Password()
```

This way it behaves just like the postgres function.

johanbrandhorst reviewed (changes requested):
Thanks for the update and for getting the test to pass, great job! I think we can make it just a little by nicer still :)

johanbrandhorst reviewed (approved):
Just one small suggestion. Make sure to use a merge commit when merging this so the original author gets commit credit (not a squash).

johanbrandhorst commented on testing/internal/e2e/tests/base/target_tcp_connect_mysql_test.go:
```diff
@@ -0,0 +1,127 @@
+	// Wait for MySQL to be ready
+	err = pool.Retry(func() error {
+		return exec.CommandContext(ctx, "docker", "exec", hostname,
+			"mysql", "-u"+user, "-p"+pw, "-e", "SELECT 1").Run()
```
🤔 [The page I linked](https://dev.mysql.com/doc/refman/8.4/en/connecting-using-uri-or-key-value-pairs.html#connecting-using-uri) in my previous comment says that "The MySQL shell" uses `--uri` for these. I don't know if that means the CLI, if it doesn't then sure lets leave it this way 😂.
